### PR TITLE
Hcalder deng 611 default privilege resource

### PR DESCRIPTION
**Pull request by calder-holt**
Updating provider to include default privileges: 

- Create resource for default privilege, requires a user (whose default privileges will be altered, ie `dbt`), a schema (to alter the privileges in), and a group (who the privileges are granted to)
- Create data source for group
- Create data source for user
- Update readme with examples of how to use resources

### CODE REVIEW

File: redshift/resource_redshift_default_privileges.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"group_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"user_id": {
```
Comment (delwaterman) [inline redshift/resource_redshift_default_privileges.go]: What do you think of naming this `table_owner_user_id`, `default_privileges_user_id`, `subject_user_id` or something like that. While `user_id` describe the type of data it does not describe what you are going to do with the user.

File: redshift/resource_redshift_default_privileges.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+		Delete: resourceRedshiftDefaultPrivilegesDelete,
+		Read:   resourceRedshiftDefaultPrivilegesRead,
+
+		Schema: map[string]*schema.Schema{
+			"schema_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"group_id": {
```
Comment (delwaterman) [inline redshift/resource_redshift_default_privileges.go:25]: `target_group_id` ?

File: redshift/resource_redshift_default_privileges.go [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+	if txErr != nil {
+		panic(txErr)
+	}
+
+	privilegesSet := validateGrants(d)
+
+	// if len(privilegesSet) == 0 {
+	// 	tx.Rollback()
+	// 	return NewError("Must have at least 1 privilege")
+	// }
```
Comment (delwaterman) [inline redshift/resource_redshift_default_privileges.go]: Kill this comment

File: redshift/resource_redshift_default_privileges.go [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func redshiftDefaultPrivilege() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceRedshiftDefaultPrivilegesCreate,
+		Update: resourceRedshiftDefaultPrivilegesCreate,
```
Comment (delwaterman) [inline redshift/resource_redshift_default_privileges.go]: Do we need this if everything is `ForceNew`

File: redshift/resource_redshift_default_privileges.go [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+		&referencesPrivilege); err != nil && err != sql.ErrNoRows {
+		tx.Rollback()
+		return err
+	}
+
+	d.Set("select", selectPrivilege)
+	d.Set("insert", insertPrivilege)
+	d.Set("update", updatePrivilege)
+	d.Set("delete", deletePrivilege)
+	d.Set("references", referencesPrivilege)
```
Comment (delwaterman) [inline redshift/resource_redshift_default_privileges.go:291]: Are we missing usage?

File: redshift/resource_redshift_schema_group_privilege.go [hunk 6]
```diff
@@ -415,57 +390,17 @@ func updatePrivilege(tx *sql.Tx, d *schema.ResourceData, attribute string, privi
...
-		for rows.Next() {
-			var tableOwner string
-			err := rows.Scan(&tableOwner)
-			if err != nil {
-				return err
-			}
-			if _, err := tx.Exec("ALTER DEFAULT PRIVILEGES FOR USER " + tableOwner + " IN SCHEMA " + schemaName + " REVOKE " + privilege + " TO GROUP " + groupName); err != nil {
-				return err
-			}
-		}
```
Comment (delwaterman) [inline redshift/resource_redshift_schema_group_privilege.go:462]: Does this mean when we ship this provider we need to add default privileges for all our current permissions grant?

**Review by delwaterman — CHANGES_REQUESTED** [review]
Nitpicks and questions.

File: redshift/resource_redshift_schema_group_privilege.go (hunk 6 above)
Comment (calder-holt) [reply redshift/resource_redshift_schema_group_privilege.go:462]: Only for dbt and fivetran

File: redshift/resource_redshift_default_privileges.go (hunk 5 above)
Comment (calder-holt) [reply redshift/resource_redshift_default_privileges.go:291]: usage is a schema permission so doesn't apply to this specific set of permissions

**Review by delwaterman — APPROVED** [review]
🚀

**Pull request merged**